src/file_cache_row.cpp

```diff
@@ -12,8 +13,8 @@
 
 file_cache_row::file_cache_row(): td(-1),status(INVALID),size(0),available_bytes(0),n_clients(0),startOffset(0),modified(false),handle(0)
 {
-	char filename[] = "/tmp/mega.XXXXXX";
-	close (mkstemp(filename));
+	std::string filename = Config::getInstance()->CACHEPATH + "/mega.XXXXXX";
+	close (mkstemp(&filename[0]));
```

Comment:
Well… possibly ;-)
I did what I could with my Java knowledge and a fair bit of research+help.
Feel free to submit corrections :-)
However, this pull-request is closed, as it has been merged into MegaFuse already. So I suppose you’ll have to create another pull-request.
Cheers,